googleapi/googleapi.go

```diff
@@ -181,6 +180,22 @@ func CheckResponseWithBody(res *http.Response, body []byte) error {
 	}
 }
 
+// errorReplyFromBody attempts to get the error from body. The body
+// may be a JSON object or JSON array, or may be something else.
+func errorReplyFromBody(body []byte) (*errorReply, error) {
+	jerr := new(errorReply)
```

Comment:
Probably still want something like this:
```suggestion
        if len(body) == 0 {
          return nil, nil
        }
	jerr := new(errorReply)
```

Reply:
Given the conditional above (`if err == nil && jerr.Error != nil {`), we need to return non-nil `err` if `errorReply` is nil. I think it's covered by `Unmarshal` if body is empty.

Reply:
Done, thanks!